qiskit/transpiler/passes/synthesis/unitary_synthesis.py

```diff
@@ -398,7 +505,13 @@ def run(self, unitary, **options):
         else:
             decomposer1q = None
 
-        decomposer2q = _basis_gates_to_decomposer_2q(basis_gates, pulse_optimize=pulse_optimize)
+        preferred_direction = None
+        if target:
+            decomposer2q, preferred_direction = self._find_decomposer_2q_from_target(
+                target, qubits, pulse_optimize
+            )
```

Comment:
`Target` defines `__len__`, so the null target (with no gates) will be Boolean-false.  It's conceivably valid to fall through to using basis gates, but I think the contract "`target` overrides `basis_gates`" means we should probably still try and use the empty `Target`, even though it'll likely fail somewhere else.

```suggestion
        if target is not None:
            decomposer2q, preferred_direction = self._find_decomposer_2q_from_target(
                target, qubits, pulse_optimize
            )
```